### ISSUE TITLE
Canonical mode – map QWERTY to Colemak

### ISSUE THREAD

**Issue by Abdillah** [jooize/vim-colemak#5]
I push my two cents here.

TLDR; A key mapping based on purely QWERTY to Colemak key placement position.
e.g. <kbd>hjkl</kbd> into <kbd>hnei</kbd>, <kbd>qwerty</kbd> into <kbd>qwfpg</kbd>, so insert mode <kbd>i</kbd> will be <kbd>u</kbd>

Vim tutorial already lots out there. And in Vim, keys are fundamental. This plugin actually works great. But, the plugin mapping actually makes more pain when learning new shortcut based on a tutorial. Additionally, when we switch to vim without plugin or computer without Colemak, the keys messed up (actually, it's my brain which does messed up).

It's great to see a colemak mapping which maps directly it's key placement from QWERTY origin into Colemak. Well, I call it canonical mode ;)

**Comment by jooize** [comment]
I want this.

**Comment by Abdillah** [comment]
I think of simple plugin option, just like the one I tried [here](https://github.com/Abdillah/vim-colemak/tree/abd-canon-switcher). Those are a simple one on one key remap (and no serious testing).

Do you think, it is possible to make a remap on runtime?
PS: My VimL knowledge isn't so good actually.

**Comment by LaserMoai** [comment]
[Coleremak](https://github.com/unlmtd/coleremak.vim) seems to fit the bill, but the colon remap at least is kind of weird.

**Comment by jamischarles** [comment]
I recently had use vim on a server and tried to see how easily I could use it by switching the keyboard to normal mode (non-colemak). 

I love colemak and this plugin is great (have been using it for several years) but I couldn't help but feel that 
1) : is so much nicer placed for vim in normal layout
2) using y for yank and p for paste seems more intuitive. It makes the keys make sense from a mnemonic perspective and not just muscle memory.

**Comment by LaserMoai** [comment]
Do you mean you like the vanilla Colemak placement? Index finger for hjkl seems nuts.

I've fixed the Tab key in Coleremak in [my fork](https://github.com/GeneralFailer/coleremak.vim) if anyone cares.

**Comment by jooize** [comment]
> 1. : is so much nicer placed for vim in normal layout
> 2. using y for yank and p for paste seems more intuitive. It makes the keys make sense from a mnemonic perspective and not just muscle memory.

I have colon on `Shift–.`, and I kind of agree about yank and paste, but where to place movements? Current approach is consistently different.

Use of remote stock Vim is awful with `hjkl` on QWERTY's `hynu`. I'll have to try Coleremak. Anything that doesn't work as expected with it?

**Comment by LaserMoai** [comment]
I haven't noticed anything wrong in my version yet.

EDIT: Ah, so @jamischarles was talking about the colon.

**Comment by jamischarles** [comment]
@GeneralFailer No, I use @jooize 's excellent movement mappings, with the exception that I swapped the `u` and `y` keys. It always felt off to me until I did that.

I tried swapping `oO` and `;:` keys in normal mode but ran into lots of issues. I currently settled on having `O` mapped to `:` in normal mode because I rarely use `O` to insert, but I like the idea of using something like `Shift-.` as @jooize mentioned above, because I occasionally get confused. My brain expects the keys to be in the same place across modes for some reason haha.

**Comment by LaserMoai** [comment]
I'm pretty sure Coleremak also breaks functionality of _emmet-vim_, so it probably needs to be rewritten. Oh well...

**Comment by LaserMoai** [comment]
Heads up,  I don't really care to fix Coleremak, so I'm leaving it unmaintained.
